_posts/2023-08-22-authz-basics.md

```diff
@@ -0,0 +1,159 @@
+
+1. User Accesses Web Application: A user attempts to access a protected resource or service within a web application.
+2. Authentication Required: Recognizing that the user is not yet authenticated, the application redirects the user to a login page.
+3. User Authenticates: The user provides their credentials, such as a username and password.
+4. Token Granted: Upon successful authentication, the identity provider issues a token to the user. This token acts as a proof of the user's identity and might contain claims, roles, or other pertinent information.
+5. Token Utilized: The web application uses this token to grant the user access to its services and to communicate with backend APIs. It is essential to note that the backend verifies the integrity of this token to ensure it's genuine and hasn't been tampered with.
+
+
+The flow above is reminiscent of the OAuth2 "Authorization Code Flow", albeit in a simplified manner. It primarily aims to illustrate the core concept of authentication. For those looking for an in-depth dive into OAuth2, consider exploring [Auth0's comprehensive guide on the Authorization Code Flow](https://auth0.com/docs/get-started/authentication-and-authorization-flow/authorization-code-flow).
+
+# Authorization Models
+
+In the intricate world of authorization, one size seldom fits all. As applications grow and requirements become more nuanced, the mechanisms to decide who gets to do what need to evolve as well. The diversity of use-cases and the need for granularity have given rise to various models of authorization. These models serve as conceptual frameworks, providing structured methods to grant or deny access to resources based on different criteria. From the rudimentary setups that hinged primarily on lists to the more intricate designs that account for dynamic attributes and relationships, the evolution of these models mirrors the increasing complexity of our digital ecosystems.
+
+Let's dive deeper into some of the prevailing models in authorization:
+
+* ACL (Access Control List): Stemming from its historical roots in the Unix file system, ACLs provide a list of permissions attached to an object. They are inherently a means to define which user have access to specific objects and the operations they can perform on them.
+
+* RBAC (Role-Based Access Control): A more structured approach, RBAC assigns permissions to specific roles rather than individual users. Users are then assigned to roles. This model offers simplicity and scalability, especially in larger systems where defining permissions for each user can become unwieldy.
+
+* ABAC (Attribute-Based Access Control): A more dynamic approach, ABAC bases access decisions on policies derived from various attributes—be it of the user, the resource, or even the surrounding environment. For instance, a document might only be accessible during working hours or if a user is located in a specific country. This approach provides nuanced, context-aware authorization.

```

Comment:
for each of those, it would be nice to see a very simple one-sentence example. something less abstract. or maybe something like: This would be most appropriate for a Twitter-Like/Multipalyer Game/Finance Administration/Planning Software/...

Reply:
Good point, I will try to find good examples.

Reply:
Added examples together with the new article.